Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp

```diff
@@ -343,15 +343,16 @@ FlushFormatPlotfile::WriteParticles(const std::string& dir,
         int_flags.resize(pc->NumIntComps(), 1);
 
         pc->ConvertUnits(ConvertDirection::WarpX_to_SI);
+        auto rtmap = pc->getParticleComps();
 
         RandomFilter const random_filter(particle_diags[i].m_do_random_filter,
                                          particle_diags[i].m_random_fraction);
         UniformFilter const uniform_filter(particle_diags[i].m_do_uniform_filter,
                                            particle_diags[i].m_uniform_stride);
         ParserFilter parser_filter(particle_diags[i].m_do_parser_filter,
-                                   compileParser<ParticleDiag::m_nvars>
+                                   compileParser<9>
```

Comment:
is ParticleDiag::m_nvars = 8?
If that changes upstream, this is kind of tricky to ensure that it will always be 9
I understand that this value needs to be known at compile time
May be creating a new static constexpr int in ParticleDiag called 
m_n_parservars = 9 and writing a comment there saying why? i.e., t,x,y,z,ux,uy,uz,upstream,tracker

Reply:
`ParticleDiag::m_nvars = 7` though it doesn't look like it's used anywhere else so it could be set to 9.